lua/nvim-tree/notify.lua

```diff
@@ -0,0 +1,44 @@
+local notify = {}
+
+local config = {
+  default_level = vim.log.levels.INFO,
+}
+
+local modes = {
+  { name = "trace", level = vim.log.levels.TRACE },
+  { name = "debug", level = vim.log.levels.DEBUG },
+  { name = "info", level = vim.log.levels.INFO },
+  { name = "warn", level = vim.log.levels.WARN },
+  { name = "error", level = vim.log.levels.ERROR },
+}
+
+do
+  local has_notify, notify_plugin = pcall(require, "notify")
```

Comment:
#1697 will fit nicely in here.

Reply:
> threshold was proposed: https://github.com/nvim-tree/nvim-tree.lua/issues/1502#issuecomment-1214289445

do you mean we should use `threshold` instead?

> https://github.com/nvim-tree/nvim-tree.lua/issues/1697 will fit nicely in here.

yeah, I had actually wanted to make the use of nvim-notify to be optional as well, but couldn't settle on a intuitive name for the option so I skipped it.

maybe something like this
```lua
  notify = {
	---@type function|string
	handler = vim.notify,
    ---@type enum
    default_level = vim.log.levels.INFO,
  },
```
tho I still think it's better to leave it for a separate PR :)